vllm/model_executor/layers/attention.py

```diff
@@ -53,13 +55,21 @@ def __init__(self, num_heads: int, head_size: int, scale: float) -> None:
             raise ValueError(f"head_size ({self.head_size}) is not supported. "
                              f"Supported head sizes: {_SUPPORTED_HEAD_SIZES}.")
 
+    def set_attn_bias(self, input_metadata: InputMetadata) -> None:
+        if input_metadata.attn_bias:
+            # Already set by a previous layer.
```

Comment:
Why do you choose this design, instead of explicitly initializing `attn_bias` in advance, say at the beginning of the `forward` function of BLOOM?

Reply:
Good question. It's because `alibi_slopes` is stored in the attention layer. If we want to create the attention bias in `BloomForCausalLM`, then we have to store `alibi_slopes` in both places (because `alibi_slopes` is also used for the decoding attention).

I kinda agree that this design is not ideal. But couldn't find a better way to do so.